crates/wasm-conventions/src/lib.rs

```diff
@@ -148,3 +151,68 @@ pub fn get_or_insert_start_builder(module: &mut Module) -> &mut FunctionBuilder
         .unwrap_local_mut()
         .builder_mut()
 }
+
+pub fn insert_target_feature(module: &mut Module, new_feature: &str) {
+    // Taken from <https://github.com/bytecodealliance/wasm-tools/blob/f1898f46bb9d96f0f09682415cb6ccfd6a4dca79/crates/wasmparser/src/limits.rs#L27>.
+    assert!(new_feature.len() <= 100_000);
+
+    // Try to find an existing section.
+    let section = module
+        .customs
+        .iter_mut()
+        .find(|(_, custom)| custom.name() == "target_features");
+
+    // If one exists, check if the target feature is already present.
+    let section = if let Some((_, section)) = section {
+        let section: &mut RawCustomSection = section.as_any_mut().downcast_mut().unwrap();
+        let mut reader = BinaryReader::new(&section.data);
+        // The first integer contains the target feature count.
+        let count = reader.read_var_u32().unwrap();
+
+        // Try to find if the target feature is already present.
+        if count != 0 {
+            for _ in 0..count {
+                // First byte is the prefix.
+                let prefix_index = reader.current_position();
+                let prefix = reader.read_u8().unwrap() as u8;
+                // Read the feature.
+                let feature = reader.read_string().unwrap();
```

Comment:
Well the tool convention says "bytes" actually and tracking down what LLD does it ends up at [this constructor](https://github.com/llvm/llvm-project/blob/aa4069ea96e5eb62bc8c7895b9d920f129611b3a/llvm/include/llvm/ADT/StringRef.h#L95-L97), which doesn't actually check for UTF-8, so I'm leaning towards just reading bytes here instead of a string.

Reply:
Done.